ckanext/datastore/logic/action.py

```diff
@@ -141,6 +142,14 @@ def datastore_create(context, data_dict):
     except db.InvalidDataError as err:
         raise p.toolkit.ValidationError(str(err))
 
+    # Set the datastore_active flag on the resource if necessary
+    if not resource_dict:
+        resource_dict = p.toolkit.get_action('resource_show')(
+            context, {'id': data_dict['resource_id']})
+    if not resource_dict.get('datastore_active'):
+        resource_dict['datastore_active'] = True
+        p.toolkit.get_action('resource_update')(context, resource_dict)
```

Comment:
Could use `resource_patch(id=data_dict['resource_id'], datastore_active=True)` here and save the extra action call. Same for datastore_delete below